internal/controller/postgrescluster/apply_test.go

```diff
@@ -147,6 +147,9 @@ func TestServerSideApply(t *testing.T) {
 	})
 
 	t.Run("StatefulSetPodTemplate", func(t *testing.T) {
+		if serverVersion.LessThan(version.MustParseGeneric("1.22.01")) {
+			t.Skip("Statefulset.Status added field in 1.22.01, cannot test `patch` in earlier versions")
```

Comment:
Yes, that version number is strange, and I don't love skipping tests in two supported version (k8s 20 and 21), but: looks like in api 0.24.2, the `availableReplicas` field is optional, but not `omitemtpy`, so we are sending a `0`; the error we get here is from line 174-5 below (the `before` is the sts intent):
```
assert.NilError(t, cc.Patch(ctx, before, client.Apply, client.ForceOwnership, reconciler.Owner))
```
And the error we get is
```
assertion failed: error is not nil: failed to create typed patch object: .status.availableReplicas: field not declared in schema
```

Reply:
> the error we get here is from line 174-5 below (the `before` is the sts intent):
> 
> ```
> assert.NilError(t, cc.Patch(ctx, before, client.Apply, client.ForceOwnership, reconciler.Owner))
> ```

I see. So the ... plain/upstream apply can't setup the test.

🤔 The TODO you've added a few lines down is fair as well. We don't need this code since we've stopped supporting 1.18. I'll try something out.